curvefs/src/client/inode_cache_manager.cpp

```diff
@@ -107,107 +107,120 @@ class TrimICacheAsyncDone : public MetaServerClientDone {
     std::shared_ptr<InodeCacheManagerImpl> inodeCacheManager_;
 };
 
-CURVEFS_ERROR InodeCacheManagerImpl::GetInode(uint64_t inodeId,
-    std::shared_ptr<InodeWrapper> &out) {
+#define GET_INODE_REMOTE(FSID, INODEID, OUT, STREAMING)                        \
+    MetaStatusCode ret = metaClient_->GetInode(FSID, INODEID, OUT, STREAMING); \
+    if (ret != MetaStatusCode::OK) {                                           \
+        LOG_IF(ERROR, ret != MetaStatusCode::NOT_FOUND)                        \
+            << "metaClient_ GetInode failed, MetaStatusCode = " << ret         \
+            << ", MetaStatusCode_Name = " << MetaStatusCode_Name(ret)          \
+            << ", inodeid = " << inodeId;                                      \
+        return MetaStatusCodeToCurvefsErrCode(ret);                            \
+    }
+
+#define PUT_INODE_CACHE(INODEID, INODEWRAPPER)                                 \
+    std::shared_ptr<InodeWrapper> eliminatedOne;                               \
+    bool eliminated = iCache_->Put(INODEID, INODEWRAPPER, &eliminatedOne);     \
+    if (eliminated) {                                                          \
+        VLOG(3) << "GetInode eliminate one inode, ino: "                       \
+                << eliminatedOne->GetInodeId()                                 \
+                << ", iCache does not evict inodes via put interface ";        \
+        assert(0);                                                             \
+    }
+
+#define REFRESH_DATA_REMOTE(OUT, STREAMING)                                    \
+    CURVEFS_ERROR rc = RefreshData(OUT, STREAMING);                            \
+    if (rc != CURVEFS_ERROR::OK) {                                             \
+        return rc;                                                             \
+    }
+
+CURVEFS_ERROR
+InodeCacheManagerImpl::GetInode(uint64_t inodeId,
+                                std::shared_ptr<InodeWrapper> &out) {
     NameLockGuard lock(nameLock_, std::to_string(inodeId));
+    // get inode from cache
     bool ok = iCache_->Get(inodeId, &out);
     if (ok) {
-        // if enableCto, we need and is unopen, we need reload from metaserver
-        if (curvefs::client::common::FLAGS_enableCto && !out->IsOpen()) {
-            VLOG(6) << "InodeCacheManagerImpl, GetInode: enableCto and inode: "
-                    << inodeId << " opencount is 0";
-            iCache_->Remove(inodeId);
-        } else {
-            curve::common::UniqueLock lgGuard = out->GetUniqueLock();
-            if (out->NeedRefreshData()) {
-                CURVEFS_ERROR rc = out->RefreshS3ChunkInfo();
-                if (rc != CURVEFS_ERROR::OK) {
-                    LOG(ERROR) << "RefreshS3ChunkInfo failed, retCode = " << rc;
-                    return rc;
-                }
-            }
-            return CURVEFS_ERROR::OK;
-        }
+        curve::common::UniqueLock lgGuard = out->GetUniqueLock();
```

Comment:
Need refreshInode here If cto enabled?

Reply:
As mentioned in [issu](https://github.com/opencurve/curve/issues/1213)，when enableCto, fuseClient.attrTimeOut and  fuseClient.entryTimeOut will be set to 0. So opening the file involves calling: 
```
1. FuseOpGetAttr ino = 1
2. FuseOpLookup parent: 1, name: test62
3. FuseOpGetAttr ino = 7340752
4. FuseOpOpen, ino: 7340752
```
Among them, only 2 and 3 calls will return attr, so just update length and s3info in one of the interface calls. This modification selection is updated in `FuseOpGetAttr`